contracts/validators/CredentialAtomicQueryV3Validator.sol

```diff
@@ -170,7 +170,10 @@ contract CredentialAtomicQueryV3Validator is CredentialAtomicQueryValidatorBase
     function _checkAuth(uint256 userID, address ethIdentityOwner) internal view {
         require(
             userID ==
-                GenesisUtils.calcIdFromEthAddress(getState().getDefaultIdType(), ethIdentityOwner),
+                GenesisUtils.calcIdFromEthAddress(
+                    getState().checkSupportedIdTypeForId(userID),
```

Comment:
Better move getting of idType outside of the calcIdFromEthAddress.

Reply:
why?